atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/SqliteOracleAdapter.java

```diff
@@ -0,0 +1,309 @@
+        return connection;
+    }
+
+    private String getConnectionUrl() {
+        return "jdbc:sqlite:" + sqliteFileLocation.getAbsolutePath();
+    }
+
+    private String generatePrefixedTableName(String prefix, TableReference tableReference) {
+        OracleDdlConfig config = ImmutableOracleDdlConfig.builder()
+                .tablePrefix(prefix)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .build();
+
+        OracleTableNameGetter tableNameGetter = OracleTableNameGetter.createDefault(config);
+        return tableNameGetter.getPrefixedTableName(tableReference);
+    }
+
+    /**
+     * A lightweight wrapper around {@link SimpleTimedSqlConnectionSupplier} that marshals some queries containing
+     * Oracle specific keywords such as PURGE to sqlite compatible queries
+     */
+    private static class SqliteSqlConnectionSupplier implements SqlConnectionSupplier {
+        private final SqlConnectionSupplier delegate;
+
+        SqliteSqlConnectionSupplier(ConnectionSupplier delegate) {
+            this.delegate = new SimpleTimedSqlConnectionSupplier(delegate);
+        }
+
+        @Override
+        public SqlConnection get() {
+            return marshalOracleQueryToSqlite(delegate.get());
+        }
+
+        @Override
+        public void close() throws PalantirSqlException {
+            delegate.close();
+        }
+
+        private SqlConnection marshalOracleQueryToSqlite(SqlConnection sqlConnection) {
+            SqlConnection spy = spy(sqlConnection);
+            doAnswer(invocation -> {
+                        String sql = invocation.getArgument(0);
+                        String strippedSql = StringUtils.removeEnd(sql, "PURGE");
+                        sqlConnection.executeUnregisteredQuery(strippedSql);
+                        return null;
+                    })
+                    .when(spy)
+                    .executeUnregisteredQuery(endsWith("PURGE"), any());
+            return spy;
+        }
+    }
+
+    /**
+     * TableDetails enables us to provide tests with sufficient physical information such that they can verify that
+     * everything that _should_ be cleaned up has been cleaned up.
+     *
+     * Generally, tests will be working on a collection of tables and will want to compare against the currently
+     * persisted table names. Thus, this also provides some utility methods for mapping from a collection of
+     * TableDetails down to a set of a particular constituent part
+     */
+    @Value.Immutable
+    interface TableDetails {
+        String reversibleTableName();
+
+        String physicalTableName();
+
+        TableReference tableReference();
+
+        static ImmutableTableDetails.Builder builder() {
+            return ImmutableTableDetails.builder();
+        }
+
+        static Set<String> mapToReversibleTableNames(Collection<TableDetails> tableDetails) {
```

Comment:
Are we certain we want a Set as the output (does it matter that we don't have the mapping of which became which and/or handle duplicates)?

Reply:
Yes, the result is intended to be compared against the results of listAllTablesWithMetadata e.t.c, which have to return sets. At least in my case, I never care about the actual mapping from a given table detail to a given prefixed table name.

Reply:
Moved to using a Set input to make it extra clear